pubtools/_impl/tracing.py

```diff
@@ -0,0 +1,138 @@
+"""Instrument functions.
+
+Usage:
+    @instrument_func()
+      def func():
+          pass
+
+"""
+import functools
+import logging
+import os
+import threading
+
+from opentelemetry import baggage, context, trace
+from opentelemetry.baggage.propagation import W3CBaggagePropagator
+from opentelemetry.propagate import set_global_textmap
+from opentelemetry.sdk.resources import SERVICE_NAME, Resource
+from opentelemetry.sdk.trace import TracerProvider
+from opentelemetry.sdk.trace.export import (BatchSpanProcessor,
+                                            ConsoleSpanExporter)
+from opentelemetry.trace import Status, StatusCode
+from opentelemetry.trace.propagation.tracecontext import \
+    TraceContextTextMapPropagator
+
+from pubtools.pluggy import pm
+
+propagator = TraceContextTextMapPropagator()
+baggage_propagator = W3CBaggagePropagator()
+log = logging.getLogger(__name__)
+
+
+def get_trace_wrapper():
+    """return a global trace wrapper instance"""
+    return TracingWrapper()
+
+
+class TracingWrapper:
+    """Wrapper class to initialize opentelemetry instrumentation and provide a helper function
+    for instrumenting a function"""
+
+    __instance = None
+
+    def __new__(cls):
+        if (
+            TracingWrapper.__instance is None
+            and os.getenv("OTEL_TRACING", "").lower() == "true"
+        ):
+            log.info("Creating TracingWrapper instance")
+            cls.__instance = super().__new__(cls)
+            exporter = pm.hook.otel_exporter()
+            if not exporter:
+                exporter = ConsoleSpanExporter()
```

Comment:
You should be able to cheat and write it like this if you don't want to bother with explicitly covering the branch codecov is complaining about:

```python
exporter = pm.hook.otel_exporter() or ConsoleSpanExporter()
```

Reply:
Thanks!